include/fmt/core.h

```diff
@@ -391,8 +391,11 @@ FMT_CONSTEXPR typename std::make_unsigned<Int>::type to_unsigned(Int value) {
 FMT_MSC_WARNING(suppress : 4566) constexpr unsigned char micro[] = "\u00B5";
 
 constexpr bool is_utf8() {
-  return FMT_UNICODE ||
-         (sizeof(micro) == 3 && micro[0] == 0xC2 && micro[1] == 0xB5);
+  // avoid buggy sign extensions in MSVC's constant evaluation mode
+  // https://developercommunity.visualstudio.com/t/C-difference-in-behavior-for-unsigned/1233612
+  using uchar = unsigned char;
+  return FMT_UNICODE || (sizeof(micro) == 3 && uchar{micro[0]} == 0xC2 &&
+                         uchar{micro[1]} == 0xB5);
```

Comment:
I suggest using a function-style cast (`uchar(...)`) for consistency with the rest of the library which doesn't use braced initialization for primitive types.